flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/AbstractCollectResultBuffer.java

```diff
@@ -128,6 +128,9 @@ private void addResults(CollectCoordinationResponse response, long responseOffse
         if (!results.isEmpty()) {
             // response contains some data, add them to buffer
             int addStart = (int) (offset - responseOffset);
+            if (addStart > results.size()) {
+                return;
+            }
```

Comment:
missing unit test?

I'm not sure if I understand this fix, maybe unit test will help me understand?

Reply:
I added a unit test.